### PR TITLE
fix(TextInputV2): remove some props and add some others

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Enhancement
- Documentation

#### What is expected?

- Fix `TextInputV2` to implement more props and remove `iconName`
- Add documentation to migrate `TextInput` to `TextInputV2`

### CODE REVIEW

File: packages/ui/src/components/TextInputV2/index.tsx [hunk 1]
```diff
@@ -133,9 +132,8 @@ type TextInputProps = {
   onChange?: (newValue: string) => void
   prefix?: ReactNode
   size?: TextInputSize
-  success?: string
+  success?: string | boolean
```
Comment (fabienhebert) [inline packages/ui/src/components/TextInputV2/index.tsx:135]: new spec ? not checking the type could lead to not wanted TextComponent as helper

Comment (matthprost) [reply packages/ui/src/components/TextInputV2/index.tsx:135]: Yep new specs, I will add a verification for the text part

**Review by Lawndlwd — APPROVED** [review]

**Review by BABAK0T0 — APPROVED** [review]

**Review by JulienSaguez — APPROVED** [review]

File: packages/ui/src/components/TextInputV2/index.tsx [hunk 2]
```diff
@@ -133,9 +132,8 @@ type TextInputProps = {
...
   suffix?: ReactNode
-  tabIndex?: number
```
Comment (DorianMaliszewski) [inline packages/ui/src/components/TextInputV2/index.tsx:138]: We should keep that its more html compliant ?

File: packages/ui/src/components/TextInputV2/index.tsx [hunk 3]
```diff
@@ -369,7 +371,7 @@ export const TextInputV2 = forwardRef<HTMLInputElement, TextInputProps>(
             </StyledInputWrapper>
           </Tooltip>
         </div>
-        {error || success || typeof helper === 'string' ? (
+        {error || typeof success === 'string' || typeof helper === 'string' ? (
```
Comment (DorianMaliszewski) [inline packages/ui/src/components/TextInputV2/index.tsx:374]: Why not use a prop state = {variant: "", text: ""} ? Like this we can handle success, warning, error ? :thinking:

Comment (DorianMaliszewski) [reply packages/ui/src/components/TextInputV2/index.tsx:374]: Here it can lead to weird behaviour if you pass error and success

Comment (matthprost) [reply packages/ui/src/components/TextInputV2/index.tsx:374]: It won't be consistent this way as depending on which one you choose the text might be facultative or mandatory, also there will be no use case where the error and success will be passed. But let's imagine you do it we defined a specific order: loading > error > success > neutral

**Pull request merged**